sqlglot/optimizer/simplify.py

```diff
@@ -695,6 +696,29 @@ def simplify_concat(expression):
     return concat_type(expressions=new_args)
 
 
+def simplify_conditionals(expression):
+    """Simplifies expressions like IF, CASE if their condition is statically known."""
+    if isinstance(expression, exp.Case):
+        this = expression.this
+        for case in expression.args["ifs"]:
+            cond = this.eq(case.this) if this else case.this
+
+            if always_true(cond):
```

Comment:
why do you need to do this? shouldn't the simplifier already handle this?

Reply:
not really:

```python
>>> from sqlglot.optimizer import optimize
>>> optimize("if(true, 1, 2)").sql()
'CASE WHEN TRUE THEN 1 ELSE 2 END'
```

Reply:
you need to canocalizie first right?

Reply:
not sure what you mean - this is running every optimization, canonicalization included